### PR TITLE
Add admin notice to warn about deactivating WC Subscriptions

### LINKED ISSUE #2963: WCPay Subscriptions: Add a notice/interaction when activating and deactivating the Subscription plugin re upgrading/downgrading
### Description

Add a "Cool your jets" notice when store admins activate and deactivate the Subscription plugin to inform them about what processes occur when upgrading/downgrading. Link off to further documentation about the processes.

### Acceptance criteria

* Intercept the activate and deactivate subscriptions plugin links.
* Display a notice with messaging about the upgrade/downgrade process. With _Continue_, _Go back_, type options.

### Designs

Needs design.

### Testing instructions

1. Go to Plugins on your admin dashboard. 
2. Activate or deactivate the WC Subscriptions plugin.
4. Observe expected behaviour - admin notice as described above.

### PR DESCRIPTION
**Pull request by james-allan**
#### Changes proposed in this Pull Request

This PR contains the changes to introduce a confirmation admin notice when users click the deactivate WC Subscriptions plugin.

Project: pdjTHR-35-p2
Notice feedback discussion: pdjTHR-aF-p2 

Screen Shot 2021-10-07 at 3 53 14 pm

#### Testing instructions

1. Install and activate the WC Subscriptions plugin.
2. Make sure the WCPay Subscriptions feature is switched on via **WCPay Dev tools**. https://d.pr/HjtrhT
3. Go to the **Plugins** admin screen. 
4. Click the link to **Deactivate** WooCommerce Subscriptions. 
5. On this branch you should be prompted with a popup notice.
6. Clicking **Yes, deactivate WooCommerce Subscriptions** will deactivate the plugin, cancel will close the modal and subscriptions will remain active.

If you disable the WCPay Subscriptions feature via **WCPay Dev tools**, deactivating the plugin doesn't lead to this prompt. The result of this means that if your site isn't eligible for WC Pay Subscriptions (feature flag isn't on) then this prompt isn't shown - the plugin just deactivates as normal. 

**Important note:** WCPay Subscriptions will be enabled by default on eligible stores. Later we'll introduce a setting to disable it.

The learn more links to a doc that hasn't been written but the URL was suggested by Brent in pdjTHR-aF-p2 

-------------------

- [ ] Added changelog entry (or does not apply)
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Added testing instructions to the [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) (or does not apply)

closes #2963

### CODE REVIEW

**Comment by james-allan** [comment]
I'm unsure how to write unit tests for this one. There's a single PHP class with 3 functions. 1 is private and the other two add scripts or the modal template.

Any advice you have would be awesome!

**Comment by haszari** [comment]
> I'm unsure how to write unit tests for this one. There's a single PHP class with 3 functions. 1 is private and the other two add scripts or the modal template.

Tricky! It's going to be tough and generally not that useful to test the modal. 

If there's business logic to detect that the plugin has been deactivated, and it's fairly functional, you could test that perhaps. But … there isn't any such code, it's done on front end fairly trivially. 

Personally I'm ok with no tests for code like this – the most important code to unit test is stable APIs. 

Possibly related – what are we doing to capture subs getting deactivated via other means (CLI, API, etc)? I realise we can't show a modal, but can we do anything else to warn the merchant? Happy for this to be done as follow up.

**Comment by james-allan** [comment]
> what are we doing to capture subs getting deactivated via other means (CLI, API, etc)? 

yeah, this will come later. It has been briefly discussed in GH https://github.com/Automattic/woocommerce-payments/issues/2963#issuecomment-928393307 and in p2 pdjTHR-aF-p2#comment-193.

The current approach is to revisit that down the track. Essentially we could catch deactivations handled by this popup and if Subscriptions was deactivated by some other means, handle them in some other way (admin notice next time viewing the dashboard or an admin email). Not sure exactly but there are options.

File: includes/subscriptions/templates/html-subscriptions-plugin-notice.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+								esc_html__( 'By deactivating the %1$sWooCommerce Subscriptions%2$s plugin, your store will switch to use %1$sSubscriptions powered by WooCommerce Payments%2$s.', 'woocommerce-payments' ),
+								'<strong>',
+								'</strong>'
+							);
+							?>
+						</br>
+						<?php
+							printf(
+								// Translators: $1 and $2 placeholders are opening and closing strong HTML tags. $3 and $4 are opening and closing link HTML tags. $5 is an opening link HTML tag.
+								esc_html__( 'Exisiting subscriptions will %1$s%3$srenew manually%4$s%2$s - subscribers will need to log in to pay for renewal. Access to premium features will also be removed. %5$sLearn more.%4$s', 'woocommerce-payments' ),
```
Comment (chickenn00dle) [inline includes/subscriptions/templates/html-subscriptions-plugin-notice.php:33]: Typo: Existing

File: includes/subscriptions/templates/html-subscriptions-plugin-notice.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+					<button class="modal-close modal-close-link dashicons dashicons-no-alt">
+						<span class="screen-reader-text">Close modal panel</span>
+					</button>
+				</header>
+				<article>
+					<p>
+						<?php
+							printf(
+								// Translators: placeholders are opening and closing strong HTML tags.
+								esc_html__( 'By deactivating the %1$sWooCommerce Subscriptions%2$s plugin, your store will switch to use %1$sSubscriptions powered by WooCommerce Payments%2$s.', 'woocommerce-payments' ),
```
Comment (chickenn00dle) [inline includes/subscriptions/templates/html-subscriptions-plugin-notice.php:24]: Nitpick: "switch to" or "use" is redundant. We can probably remove one or the other.

...your store will switch to Subscriptions powered by..., or
...your store will use Subscriptions powered by...

**Review by chickenn00dle — CHANGES_REQUESTED** [review]
Nice work @james-allan!

I tested and this LGTM.

There is just a small typo and nitpick, then this is good to go (once the learn more doc has been published) 👍

**Comment by james-allan** [comment]
Parking this PR until we publish accompanying doc.

**Comment by haszari** [comment]
The proposed URL is `docs.woocommerce.com/document/subscriptions/deactivation/` (see docs spreadsheet/thread pbDlC7-o0-p2#comment-650). 

This PR already uses correct URL – so it's good to merge when you're ready @james-allan 🎉

**Pull request merged**